### PR TITLE
Demo.props.choices(initialValue, choices)

### PR DESCRIPTION
**Pull request by vslinko**

### CODE REVIEW

**Comment by vslinko** [comment]
Renamed `enum` to `choices` because `enum` is reserved word.

**Comment by rpominov** [comment]
I was thinking that choices could be not only strings, but any values, including complex values like components.

For instance:

``` js
<Demo target={Foo} props={{
  children: Demo.props.choices({
    baz: <Baz />,
    bar: <Bar />
  })
}} />
```

I think `Demo.props.choices()` should accept array or object as first arg, and optional `initialValue` as second (by default initial value is the first value). If object passed we use keys as labels and values as values, and in case of array we use `stringify(item)` as labels, and items as values.

What do you think?

**Comment by vslinko** [comment]
Agreed, I'll do it.

File: src/index.js [hunk 1]
```diff
@@ -65,21 +65,12 @@ export default React.createClass({
...
-      values: {
-        ...this.state.values,
-        ...changes
-      }
-    })
-  },
-
   render() {
-    const props = {...this.state.values, ...this.getCallbacks()}
+    const componentProps = {...this.state.values, ...this.getCallbacks()}
```
Comment (rpominov) [inline src/index.js:69]: jfyi, i've fixed the name conflict by renaming upper level `props`

**Comment by rpominov** [comment]
Any update on this? I'm thinking of doing it by myself at this weekend :smile:

**Comment by vslinko** [comment]
I have no time on this weekend, you could continue.

**Comment by rpominov** [comment]
Ok, no problem :)

**Comment by rpominov** [comment]
Merged your commit and finished it up in master. Thanks!

**Pull request closed without merging**